src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMModelMethods.scala

```diff
@@ -63,4 +106,13 @@ trait LightGBMModelMethods extends LightGBMModelParams {
   protected def featuresShap(features: Vector): Vector = {
     Vectors.dense(getLightGBMBooster.featuresShap(features))
   }
+
+  protected def updateBoosterParamsBeforePredict(): Unit = {
+    if (getNumIterations != LightGBMConstants.DefaultNumIterations) {
+      getModel.setNumIterations(getNumIterations)
+    }
+    if (getStartIteration != LightGBMConstants.DefaultStartIteration) {
+      getModel.setStartIteration(getStartIteration)
+    }
+  }
```

Comment:
```suggestion
  protected def updateBoosterParamsBeforePredict(): Unit = {
      getModel.setNumIterations(getNumIterations)
      getModel.setStartIteration(getStartIteration)
  }
```

Wouldnt this yield the same behavior?

Reply:
ah, I don't want to set numIterations if the best iteration was set on the booster... let me think, maybe I can clean this up a bit more... certainly for startIteration it is fine..

Reply:
I think I have a more readable solution: I set
```
new LightGBMClassificationModel(uid)
...
.setNumIterations(lightGBMBooster.bestIteration)
```
on each of the models when they are created by the learner, so I can remove this somewhat confusing check here